pkg/build/controller/strategy/custom.go

```diff
@@ -22,12 +23,22 @@ type CustomBuildStrategy struct {
 
 // CreateBuildPod creates the pod to be used for the Custom build
 func (bs *CustomBuildStrategy) CreateBuildPod(build *buildapi.Build) (*kapi.Pod, error) {
-	data, err := bs.Codec.Encode(build)
+	strategy := build.Spec.Strategy.CustomStrategy
+
+	codec := bs.Codec
+	if strategy.BuildAPIVersion != "" {
+		interfaces, err := latest.InterfacesFor(strategy.BuildAPIVersion)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get codec for requested API version (%q): %v", strategy.BuildAPIVersion, err)
```

Comment:
No, you can't. It won't go into `Failed` because the retry controller doesn't recognize it as a fatal error. We can change that. How about adding BC validation check for the version, making sure we recognize it?